extensions/ftp/src/main/java/org/mule/extension/ftp/api/sftp/SftpFileSystem.java

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.ftp.api.sftp;
+
+import static org.apache.commons.lang.StringUtils.EMPTY;
+import static org.mule.runtime.core.config.i18n.MessageFactory.createStaticMessage;
+import org.mule.extension.ftp.api.FtpConnector;
+import org.mule.extension.ftp.api.ftp.FtpFileSystem;
+import org.mule.extension.ftp.internal.sftp.command.SftpCopyCommand;
+import org.mule.extension.ftp.internal.sftp.command.SftpCreateDirectoryCommand;
+import org.mule.extension.ftp.internal.sftp.command.SftpDeleteCommand;
+import org.mule.extension.ftp.internal.sftp.command.SftpListCommand;
+import org.mule.extension.ftp.internal.sftp.command.SftpMoveCommand;
+import org.mule.extension.ftp.internal.sftp.command.SftpReadCommand;
+import org.mule.extension.ftp.internal.sftp.command.SftpRenameCommand;
+import org.mule.extension.ftp.internal.sftp.command.SftpWriteCommand;
+import org.mule.extension.ftp.internal.sftp.connection.SftpClient;
+import org.mule.runtime.api.connection.ConnectionExceptionCode;
+import org.mule.runtime.api.connection.ConnectionValidationResult;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.MuleRuntimeException;
+import org.mule.runtime.module.extension.file.api.AbstractFileSystem;
+import org.mule.runtime.module.extension.file.api.FileAttributes;
+import org.mule.runtime.module.extension.file.api.command.CopyCommand;
+import org.mule.runtime.module.extension.file.api.command.CreateDirectoryCommand;
+import org.mule.runtime.module.extension.file.api.command.DeleteCommand;
+import org.mule.runtime.module.extension.file.api.command.ListCommand;
+import org.mule.runtime.module.extension.file.api.command.MoveCommand;
+import org.mule.runtime.module.extension.file.api.command.ReadCommand;
+import org.mule.runtime.module.extension.file.api.command.RenameCommand;
+import org.mule.runtime.module.extension.file.api.command.WriteCommand;
+import org.mule.runtime.module.extension.file.api.lock.PathLock;
+import org.mule.runtime.module.extension.file.api.lock.URLPathLock;
+
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Path;
+
+import javax.inject.Inject;
+
+import org.apache.commons.net.ftp.FTPClient;
+
+/**
+ * Implementation of {@link FtpFileSystem} for files residing on a SFTP server
```

Comment:
no, it also works if you have an import to that class. It's actually what intelliJ recommends you to do